Core/src/org/sleuthkit/autopsy/discovery/search/DiscoveryEventUtils.java

```diff
@@ -203,6 +229,47 @@ public SearchCompleteEvent(Map<GroupKey, Integer> groupMap, List<AbstractFilter>
 
     }
 
+    /**
+     * Event to signal the completion of a search being performed.
+     */
+    public static final class ArtifactListRetrievedEvent {
```

Comment:
Yes you have interpreted that correctly as there may be no artifacts in the results of the search, so using it made sense to use the one event for both clearing and results since the effect of 0 results should be the same as a clear. 

I agree your name is a definite improvement and will change it.